[inductor] Fix a cpp_wrapper issue when fx_passes modified fx graph

desertfire opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #102738
* __->__ #102851

Summary: Currently cpp_wrapper for CUDA does it in two passe, which
means we need to deepcopy the input module to isolate any fx
transformations between the two passes.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225

desertfire committed:
[inductor] Fix a cpp_wrapper issue when fx_passes modified fx graph

Summary: Currently cpp_wrapper for CUDA does it in two passe, which
means we need to deepcopy the input module to isolate any fx
transformations between the two passes.

[ghstack-poisoned]

desertfire committed:
Update on "[inductor] Fix a cpp_wrapper issue when fx_passes modified fx graph"


Summary: Currently cpp_wrapper for CUDA does it in two passe, which
means we need to deepcopy the input module to isolate any fx
transformations between the two passes.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel

[ghstack-poisoned]

jansel commented on torch/_inductor/compile_fx.py:
```diff
@@ -193,7 +193,10 @@ def wrapper(gm: torch.fx.GraphModule, example_inputs: List[torch.Tensor], **kwar
+                # deepcopy(gm) makes sure no graph modification from the first pass will
+                # leak to the second pass. It does increase memory pressure, but the problem
+                # can be alleviated once we have FakeModule.
+                compiled = inner_compile(deepcopy(gm), example_inputs, **kwargs_patched)
```
Let's not use deepcopy here since it might try to copy non-FX stuff referenced by the graph.

I think you can just run torch.fx.Transformer() to shallow copy an FX graph.

desertfire committed:
Update on "[inductor] Fix a cpp_wrapper issue when fx_passes modified fx graph"


Summary: Currently cpp_wrapper for CUDA does it in two passe, which
means we need to deepcopy the input module to isolate any fx
transformations between the two passes.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel

[ghstack-poisoned]

desertfire commented:
@pytorchbot merge

desertfire commented:
@pytorchbot merge
